sdk/core/azure-core/azure/core/settings.py

```diff
@@ -72,14 +72,14 @@ def convert_bool(value):
     :raises ValueError: If conversion to bool fails
 
     """
-    if value in (True, False):
-        return value  # type: ignore
-
-    val = value.lower()  # type: ignore
-    if val in ["yes", "1", "on", "true", "True"]:
-        return True
-    if val in ["no", "0", "off", "false", "False"]:
-        return False
+    if isinstance(value, bool):
```

Comment:
I'm not happy with this change, what was the mypy error here?

Reply:
Incompatible return type - we return a bool. used a cast here instead.